types/react-relay/test/react-relay-tests.tsx

```diff
@@ -55,167 +55,268 @@ const MyQueryRenderer = (props: { name: string }) => (
 );
 
 // ~~~~~~~~~~~~~~~~~~~~~
-// Modern FragmentContainer
+// Modern RefetchContainer
 // ~~~~~~~~~~~~~~~~~~~~~
-const MyFragmentContainer = createFragmentContainer(
-    class TodoListView extends React.Component {
+
+type StoryLike = (storyID: string) => void;
+
+// Artifact produced by relay-compiler-language-typescript
+const enum _Story_story$ref {}
+type Story_story$ref = _Story_story$ref & FragmentReference;
+type Story_story = {
+    readonly id: string;
+    readonly text: string;
+    readonly isPublished: boolean;
+    readonly " $refType": Story_story$ref;
+};
+
+const Story = (() => {
+    interface Props {
+        relay: RelayRefetchProp;
+        story: Story_story;
+        onLike: StoryLike;
+        ignoreMe?: {};
+    }
+
+    interface State {
+        isLoading: boolean;
+    }
+
+    class Story extends React.Component<Props> {
+        state = {
+            isLoading: false
+        };
+
+        componentDidMount() {
+            setInterval(this.handleRefresh.bind(this), 1000);
+        }
+
+        handleRefresh() {
+            this.setState({ isLoading: true });
+            this.props.relay.refetch({ id: this.props.story.id }, {}, error => {
+                this.setState({ isLoading: false });
+            }, { force: true });
+        }
+
         render() {
-            return <div />;
+            return (
+                <div>
+                    {this.props.story.isPublished ? "" : "Draft: "}
+                    {this.props.story.text}
+                    {this.state.isLoading && <span>♺</span>}
+                    <button onClick={() => this.props.onLike(this.props.story.id)}>LIKE</button>
+                </div>
+            );
         }
-    },
-    {
-        item: graphql`
-            fragment TodoItem_item on Todo {
-                text
-                isComplete
+    }
+
+    const StoryRefetchContainer = createRefetchContainer(
+        Story,
+        {
+            story: graphql`
+                fragment Story_story on Todo {
+                    id
+                    text
+                    isPublished
+                }
+            `,
+        },
+        graphql.experimental`
+            query StoryRefetchQuery($id: ID!) {
+                story(id: $id) {
+                    ...Story_story
+                }
             }
-        `,
+        `
+    );
+
+    function doesNotRequireRelayPropToBeProvided() {
+        const onLike = (id: string) => console.log(`Liked story #${id}`);
+        const story: { " $fragmentRefs": Story_story$ref } = {} as any;
+        // TODO: Fix requirement to cast fragment reference as `any`.
```

Comment:
@kastermester I left a bunch of these TODO comments, which are the places that our future fragment reference should fix.